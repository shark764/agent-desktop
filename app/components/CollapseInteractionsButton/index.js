/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/**
*
* CollapseInteractionsButton
*
*/

import React from 'react';
import PropTypes from 'prop-types';

import Radium from 'radium';

function CollapseInteractionsButton(props) {
  const styles = {
    base: {
      borderTop: '1px solid black',
      borderRight: '1px solid black',
      borderBottom: '1px solid black',
      borderRadius: '3px',
      height: '75px',
      backgroundColor: '#072931',
      position: 'absolute',
      bottom: '50%',
      cursor: 'pointer',
      transition: 'left 0.3s linear',
      left: '0px',
      zIndex: '2',
    },
    expandedBase: {
      left: '72px',
    },
    triangle: {
      borderWidth: '6px',
      borderStyle: 'solid',
      borderColor: '#FFF transparent transparent #FFF',
      borderImage: 'initial',
      transform: 'rotate(135deg)',
      borderRadius: '3px',
      boxShadow: '-6px -6px 6px -4px rgba(0,0,0,0.29)',
      position: 'relative',
      top: '45%',
      right: '25%',
    },
    expandedTriangle: {
      transform: 'rotate(-45deg)',
      right: '0',
      left: '25%',
    },
  };

  return (
    <div
      style={[styles.base, !props.isCollapsed && styles.expandedBase]}
      onClick={props.toggleInteractionsBar}
    >
      <div
        style={[styles.triangle, !props.isCollapsed && styles.expandedTriangle]}
      />
    </div>
  );
}

CollapseInteractionsButton.propTypes = {
  isCollapsed: PropTypes.bool.isRequired,
  toggleInteractionsBar: PropTypes.func.isRequired,
};

export default Radium(CollapseInteractionsButton);
