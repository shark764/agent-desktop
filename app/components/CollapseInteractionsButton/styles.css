  
  .pendingInteraction {
	background: rgba(35, 209, 244, 0.685);
	box-shadow: 0 0 0 rgba(35,205,244, 0.4);
	animation: pendingInteraction 2s infinite;
  }
  .interactionNotPending {
	background-color: rgba(7, 41, 49, 0.75);
  }
  
  @-webkit-keyframes pendingInteraction {
	0% {
	  -webkit-box-shadow: 0 0 0 0 rgba(35,205,244, 0.4);
	}
	70% {
		-webkit-box-shadow: 0 0 0 15px rgba(35,205,244, 0);
	}
	100% {
		-webkit-box-shadow: 0 0 0 0 rgba(35,205,244, 0);
	}
  }
  @keyframes pendingInteraction {
	0% {
	  -moz-box-shadow: 0 0 0 0 rgba(35,205,244, 0.4);
	  box-shadow: 0 0 0 0 rgba(35,205,244, 0.4);
	}
	70% {
		-moz-box-shadow: 0 0 0 15px rgba(35,205,244, 0);
		box-shadow: 0 0 0 15px rgba(35,205,244, 0);
	}
	100% {
		-moz-box-shadow: 0 0 0 0 rgba(35,205,244, 0);
		box-shadow: 0 0 0 0 rgba(35,205,244, 0);
	}
  }
