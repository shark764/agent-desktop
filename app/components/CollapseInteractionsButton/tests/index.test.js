/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import CollapseInteractionsButton from '../index';

describe('<CollapseInteractionsButton />', () => {
  it('should render correctly', () => {
    const rendered = shallow(
      <CollapseInteractionsButton
        id="mockId"
        isCollapsed={false}
        toggleInteractionsBar={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
