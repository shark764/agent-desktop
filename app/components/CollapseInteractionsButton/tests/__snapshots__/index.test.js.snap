// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollapseInteractionsButton /> should render correctly 1`] = `
<div
  data-radium={true}
  id="toggleInteractionsBar"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#072931",
      "borderBottom": "1px solid black",
      "borderRadius": "3px",
      "borderRight": "1px solid black",
      "borderTop": "1px solid black",
      "bottom": "50%",
      "cursor": "pointer",
      "height": "75px",
      "left": "72px",
      "position": "absolute",
      "transition": "left 0.3s linear",
      "zIndex": "2",
    }
  }
>
  <div
    data-radium={true}
    style={
      Object {
        "borderColor": "#FFF transparent transparent #FFF",
        "borderImage": "initial",
        "borderRadius": "3px",
        "borderStyle": "solid",
        "borderWidth": "6px",
        "boxShadow": "-6px -6px 6px -4px rgba(0,0,0,0.29)",
        "left": "25%",
        "position": "relative",
        "right": "0",
        "top": "45%",
        "transform": "rotate(-45deg)",
      }
    }
  />
</div>
`;
